libs/solidity-metadata/src/lib.rs

```diff
@@ -0,0 +1,317 @@
+use minicbor::{data::Type, Decode, Decoder};
+use semver::Version;
+use std::str::FromStr;
+use thiserror::Error;
+
+/// Parsed metadata hash
+/// (https://docs.soliditylang.org/en/v0.8.14/metadata.html#encoding-of-the-metadata-hash-in-the-bytecode).
+///
+/// Currently we are interested only in `solc` value.
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct MetadataHash {
+    pub solc: Option<Version>,
+}
+
+impl MetadataHash {
+    pub fn from_cbor(encoded: &[u8]) -> Result<(Self, usize), minicbor::decode::Error> {
+        let mut context = DecodeContext::default();
+        let result = minicbor::decode_with(encoded, &mut context)?;
+
+        Ok((result, context.used_size))
+    }
+}
+
+#[derive(Debug, Error)]
+enum ParseMetadataHashError {
```

Comment:
It's common to have just `Error` type if it's imported as root (i.e. people will use this as `solidity_metadata::Error`

Reply:
also make this public?

Reply:
main function returns `minicbor::decode::Error::custom`, so you can only get Box<dyn std::error::Error> for that. For now there is no sense making this error public and therefore changing  its name